main.py

```diff
@@ -628,7 +628,32 @@ def visit_TryExcept(self, tree):
                     T('[__ctx.__enter__(), __ctx.__exit__(None, None, None), __out[0](lambda: {after})][2]')))
 
     def visit_TryFinally(self, tree):
-        raise NotImplementedError('Open problem: try-finally')
+        body = self.many_to_one(
+            tree.body, after=T('(lambda after: after())')).format(
+                pre_return=T('(lambda ret: lambda after: ret)({pre_return}'),
+                post_return=T('{post_return})'))
+        finalbody = self.many_to_one(
+            tree.finalbody, after=T('(lambda body: body)')).format(
+                pre_return=T('(lambda ret: lambda body: lambda after: ret)({pre_return}'),
+                post_return=T('{post_return})'))
+        return \
+            lambda_function({'__out': '[None]'}).format(
+                lambda_function({
+                    '__ctx': T(
+                        "{__contextlib}.nested(type('except', (), {{"
+                        "'__enter__': lambda self: None, "
+                        "'__exit__': lambda __self, __exctype, __value, __traceback: "
+                        "{finalbody}}})(), "
+                        "type('try', (), {{"
+                        "'__enter__': lambda self: None, "
+                        "'__exit__': lambda __self, __exctype, __value, __traceback: "
+                        "{body}}})())").format(
+                            body=assignment_component('False', '__out[0]', body),
+                            finalbody=assignment_component(
+                                'False', '__out[0]',
```

Comment:
This turns out to be wrong because `finally: return …` is supposed to swallow any pending exceptions.